kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLogManager.java

```diff
@@ -115,5 +115,19 @@ public CompletableFuture<Void> takeProducerStateSnapshots() {
         });
         return FutureUtil.waitForAll(handles);
     }
+
+    public CompletableFuture<Void> purgeAbortedTxns() {
```

Comment:
In my mind, an ideal solution would be to have a listener that the managed ledger calls when a ledger is deleted. Then, this protocol handler could register for events and we'd be able to avoid scheduling this loop. In the absence of this event driven solution, I think this loop makes sense.

However, it does mean that we could end up with an hours worth of aborted transactions, which could lead to issues if there are many aborted transactions.

Reply:
In any case Kafka retention is usually 2 weeks so we are going to always keep 2 weeks of Aborted Transactions in memory, because we must keep the history up to the oldest "readable" offset.

Being "reactive" is not a big deal
WDYT ?

Reply:
The primary benefits of a reactive solution are that it requires less configuration (no need to decide how frequently to poll) and the broker doesn't need to spend extra cycles polling. I agree that we can use this solution for now. I created an issue so we don't lose the potential feature: https://github.com/datastax/starlight-for-kafka/issues/33.